lib/diplomat/session.rb

```diff
@@ -3,50 +3,99 @@
 module Diplomat
   class Session < Diplomat::RestClient
 
-    @access_methods = [ :create, :destroy, :list, :renew ]
+    @access_methods = [:create, :destroy, :list, :renew, :info, :node]
 
     # Create a new session
     # @param value [Object] hash or json representation of the session arguments
+    # @param options [Hash] session options
+    # @param options [String] :dc datacenter to create session for
     # @return [String] The sesssion id
-    def create value=nil
+    def create value=nil, options=nil
       # TODO: only certain keys are recognised in a session create request,
       # should raise an error on others.
       raw = @conn.put do |req|
-        req.url "/v1/session/create"
+        url = ['/v1/session/create']
+        url += use_named_parameter('dc', options[:dc]) if options and options[:dc]
+
+        req.url concat_url url
         req.body = (if value.kind_of?(String) then value else JSON.generate(value) end) unless value.nil?
       end
       body = JSON.parse(raw.body)
-      return body["ID"]
+      body['ID']
     end
 
     # Destroy a session
     # @param id [String] session id
-    # @return [nil]
-    def destroy id
+    # @param options [Hash] session options
+    # @param options [String] :dc datacenter to destroy session for
+    # @return [String] Success or failure of the session destruction
+    def destroy id, options=nil
       raw = @conn.put do |req|
-        req.url "/v1/session/destroy/#{id}"
+        url = ["/v1/session/destroy/#{id}"]
+        url += use_named_parameter('dc', options[:dc]) if options and options[:dc]
+
+        req.url concat_url url
       end
-      return raw.body
+      raw.body
     end
     
     # List sessions
-    # @return [Struct]
-    def list
+    # @param options [Hash] session options
+    # @param options [String] :dc datacenter to list sessions
+    # @return [OpenStruct]
+    def list options=nil
       raw = @conn.get do |req|
-        req.url "/v1/session/list"
+        url = ['/v1/session/list']
+        url += use_named_parameter('dc', options[:dc]) if options and options[:dc]
+
+        req.url concat_url url
       end
-      JSON.parse(raw.body)
+      JSON.parse(raw.body).map { |session| OpenStruct.new session }
```

Comment:
This would not be a breaking change since OpenStructs can be accessed in the same manner as Hashes, i.e., some_hash['key1']['key2']. This change just makes the return value match the current documentation.